lib/messages.js

```diff
@@ -175,12 +175,21 @@ module.exports = {
             phone = utils.getHealthCenterPhone(doc);
         } else if (recipient === 'grandparent') {
             phone = utils.getDistrictPhone(doc);
-        }
-
-        if (!phone && doc.fields && doc.fields[recipient]) {
-            // try to resolve a specified property/field name
+        } else if (doc.fields && doc.fields[recipient]) {
+            // Try to resolve a specified property/field name
             phone = doc.fields[recipient];
+        } else if (doc[recipient]) {
+            // Or directly on the doc
+            phone = doc[recipient];
+        } else if (recipient.indexOf('.') > -1) {
+            // Or multiple layers by executing it as a statement
+            try {
+                phone = utils.evalExpression({doc: doc}, 'doc.' + recipient);
```

Comment:
I know we use `eval` everywhere but maybe we can come up with a safer way to walk a JSON object? For example we use [object-path](https://www.npmjs.com/package/object-path) in api.

Reply:
Good idea!